firefox-ios/Client/Frontend/Autofill/LoginListViewModel.swift

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Storage
+import Foundation
+
+@MainActor
+class LoginListViewModel: ObservableObject {
+    @Published var logins: [EncryptedLogin] = []
+
+    private let tabURL: URL
+    private let loginStorage: LoginStorage
+    private let logger: Logger
+    let onLoginCellTap: (EncryptedLogin) -> Void
+    let manageLoginInfoAction: () -> Void
+
+    var shortDisplayString: String {
+        tabURL.baseDomain ?? ""
+    }
+
+    init(
+        tabURL: URL,
+        loginStorage: LoginStorage,
+        logger: Logger,
+        onLoginCellTap: @escaping (EncryptedLogin) -> Void,
+        manageLoginInfoAction: @escaping () -> Void
+    ) {
+        self.tabURL = tabURL
+        self.loginStorage = loginStorage
+        self.logger = logger
+        self.onLoginCellTap = onLoginCellTap
+        self.manageLoginInfoAction = manageLoginInfoAction
+    }
+
+    func fetchLogins() async {
```

Comment:
Unit tests? We even have the mocks implemented already to have tests for those functions 👀